src/trans.rs

```diff
@@ -574,20 +582,20 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
             // TODO: when it's possible to print characters or interact with the environment, also
             //       handle #[lang = "panic_fmt"] to support panic messages
             let mut is_fn_panic = false;
-            if self.sig.output == ty::FnOutput::FnDiverging {
-                let attrs = self.tcx.map.attrs(nid);
-                for attr in attrs {
-                    if let MetaItemKind::NameValue(ref node, ref span) = attr.node.value.node {
-                        if node == "lang" {
-                            if let LitKind::Str(ref value, _) = span.node {
-                                if value == "panic" {
-                                    is_fn_panic = true;
-                                }
-                            }
-                        }
-                    }
-                }
-            }
+            // if self.sig.output == ty::FnOutput::FnDiverging {
```

Comment:
`self.sig.output.is_never()`. But also, this whole block is a very lengthy way to do `Some(self.did) == tcx.lang_items.panic_fn()` (which doesn't require the check for the return type).